### PR TITLE
[BE][8/N] Remove ShardedTensor from TP FSDP integration test and other tests depending on Sharded Linear

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96254

We removed ShardedLinear in https://github.com/pytorch/pytorch/pull/95948 but it broke TP_FSDP integration test because it is using ShardedTensor in the test. Migrating using DTensor fixes the test. DTensor shards the bias too so that we need to change the test a little bit.

### CODE REVIEW

**Comment by fduwjj** [comment]
Looks like the failure about dynamo benchmark is not related to this PR "No CUDA GPUs are available"

File: test/distributed/_shard/sharding_plan/test_sharding_plan.py [hunk 1]
```diff
@@ -61,200 +56,6 @@ def build_plan(self, module: nn.Module) -> ShardingPlan:
 
 
 class TestShardingPlan(ShardedTensorTestBase):
-
-    @with_comms(init_rpc=False)
-    @skip_if_lt_x_gpu(TEST_GPU_NUM)
-    @requires_nccl()
-    def test_sharding_plan_simple_megatron(self):
```
Comment (fduwjj) [inline test/distributed/_shard/sharding_plan/test_sharding_plan.py:68]: These tests are relied on Sharded Linear and we should also remove these two tests.

**Comment by huydhn** [comment]
The new periodic multigpu failure https://hud.pytorch.org/pr/96254#11842511229 also look related, so I guess it's another test to be update

**Comment by fduwjj** [comment]
@huydhn Ahhh that's no wonder why I didn't see it. We need to remove that test too. We have one already for DTensor under [test/distributed/](https://github.com/pytorch/pytorch/blob/master/test/distributed/tensor/parallel/test_tp_examples.py). Removed that test too.

**Review by huydhn — APPROVED** [review]
LGTM! You might see buck failure on periodic, but it's broken in trunk at the moment. So the failure is expected.

**Comment by huydhn** [comment]
@fduwjj I have a small follow-up PR https://github.com/pytorch/pytorch/pull/96431 to cleanup reference to the deleted test.  The multigpu test complains expectedly that it couldn't find the test.

**Pull request closed without merging**